### PR TITLE
SEP-31: add callback URL registration endpoint

### LINKED ISSUE #1209: SEP-31: add a transaction status callback mechanism
### What problem does your feature solve?

Sending Anchors with high transaction throughput must poll Receiving Anchors' `GET /transactions/:id` endpoint for many transactions concurrently in order to detect status changes in a timely manner. This is resource-intensive for both the Sending and Receiving Anchor.

### What would you like to see?

A mechanism by which Sending Anchors can specify URLs for the Receiving Anchor to make requests to when a transaction's status changes.

One idea is to add a `callback_url` argument to `POST /transactions` and `PATCH /transactions/:id`. This would allow clients to define a URL with the initial creation of a transaction using `POST` but also allow clients to update the URL used using `PATCH`.

### PR DESCRIPTION
**Pull request by JakeUrban**
resolves #1209 

Adds a `PUT /transactions/:id/callback` endpoint for Sending Anchor to request callback requests to a specified URL. This removes the need for Sending Anchors to poll the statuses of in-progress transactions, reducing the chatter between services, especially for anchors with high transaction throughput.

I considered adding this functionality into the existing `POST /transactions` and `PATCH /transactions` requests as an optional parameter, but figured that it was likely for Receiving Anchors who did not yet implement this feature to simply ignore the `callback_url` request attribute, leading Sending Anchors to believe their callback URL was accepted but never receiving the callback requests.

Note: originally we decided not to add this functionality in #636. I disagree with my former self there and don't see why we wouldn't add this capability for the benefit of Sending Anchors that wanted it. Receiving Anchors can chose not to implement this until a Sending Anchors requests it. cc @gaclaudiu

### CODE REVIEW

**Review by leighmcculloch — APPROVED** [review]
This new endpont is subject to the same authenticaton requirements of SEP-10? LGTM!

**Comment by JakeUrban** [comment]
> This new endpont is subject to the same authenticaton requirements of SEP-10?

Yes it is, but I think thats clear from the authentication section. 

I'll wait for another member of the anchor eng team to approve before merging.

**Review by marcelosalloum — COMMENTED** [review]
There seem to be a few details missing in this change:
* Add this endpoint to the [API Endpoints](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0031.md#api-endpoints) section.
* Mention the usage of this endpoint in the "detailed sending/receiving anchor flows"

**Comment by JakeUrban** [comment]
Thanks for the comments @marcelosalloum ! I definitely missed those flows.

File: ecosystem/sep-0031.md
Comment (marcelosalloum) [inline ecosystem/sep-0031.md:193]: This wording contradicts what we agreed for PR [#1203](https://github.com/stellar/stellar-protocol/pull/1203/files#diff-6abf4f55bde3bfb0f1b81e2437e70d45ea84c64213c3d09715bd901e3ec2a101R191). 

The wallet should actually double-check the amount_in from the GET /transactions/:id before sending the payment.

**Review by marcelosalloum — COMMENTED** [review]
Not so sure about one line of your change:

**Review by marcelosalloum — APPROVED** [review]

**Pull request merged**
